src/modules/launcher/Wox.Infrastructure/Storage/StoragePowerToysVersionInfo.cs

```diff
@@ -0,0 +1,111 @@
+
+            // If there is some error in populating/retrieving the version numbers, then the cache must be deleted
+            // This case will not be hit, but is present as a fail safe
+            if(String.IsNullOrEmpty(version1) || String.IsNullOrEmpty(version2))
+            {
+                return true;
+            }
+
+            string[] split1 = version1.Split( new string[] { version, period }, StringSplitOptions.RemoveEmptyEntries); 
+            string[] split2 = version2.Split( new string[] { version, period }, StringSplitOptions.RemoveEmptyEntries); 
+
+            for(int i=0; i<3; i++)
+            {
+                if(int.Parse(split1[i]) < int.Parse(split2[i]))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public string GetPreviousVersion()
+        {
+            if (File.Exists(FilePath))
+            {
+                return File.ReadAllText(FilePath);
+            }
+            else
+            {
+                // which means it's an old version of PowerToys
+                string oldVersion = "v0.0.0";
+                return oldVersion;
+            }
+        }
+
+        private string GetFilePath(String AssociatedFilePath, int type)
+        {
+            string suffix = string.Empty;
+            string cacheSuffix = ".cache";
+            string jsonSuffix = ".json";
+
+            if(type == (uint)StorageType.BINARY_STORAGE)
+            {
+                suffix = cacheSuffix;
+            }
+            else if(type == (uint)StorageType.JSON_STORAGE)
+            {
+                suffix = jsonSuffix;
+            }
+
+            string filePath = AssociatedFilePath.Substring(0, AssociatedFilePath.Length - suffix.Length) + "_version.txt";
+            return filePath;
+        }
+
+        public StoragePowerToysVersionInfo(String AssociatedFilePath, int type)
+        {
+            FilePath = GetFilePath(AssociatedFilePath, type);
+            // Get the previous version of PowerToys and cache Storage details from the CacheDetails.json storage file
+            String previousVersion = GetPreviousVersion();
+            currentPowerToysVersion = Microsoft.PowerToys.Settings.UI.Lib.Utilities.Helper.GetProductVersion();
+
+            // After this version we no longer have to delete the cache
+            // Right now, it has been set to a large value
+            // NOTE: This must be changed to the least portable version number so that we no longer delete cache after that version
+            String portableVersion = "v10.0.0";
+
+            // If the previous version is below a set threshold, then we want to delete the file
+            // However, we do not want to delete the cache if the same version of powerToys is being launched
+            if (Lessthan(previousVersion, portableVersion) && !previousVersion.Equals(currentPowerToysVersion, StringComparison.OrdinalIgnoreCase))
```

Comment:
Why not `Lessthan(prevV, currentV)`? I understand that deleting the cache is temporary and may be changed later once its shifted to JSON or something of that sort but why use 10.0.0 vs the current version?

Reply:
Because ideally we would want to have a portable version after which we do not have to worry about clearing cache. I just added it this way so that we have that goal in mind.

Reply:
But wouldn't that be equivalent to just doing <currentVersion (not <=), and whenever we make it portable (say 0.20.0), this deletion logic would be changed such that if <0.20.0 delete , else do porting logic?

Reply:
As you said both are equivalent. I changed it.